source/Helpers/POETradeAPI/Exchange/ExchangePrefetch.cs

```diff
@@ -0,0 +1,75 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Sidekick.Helpers.POETradeAPI.Exchange
+{
+    class ExchangePrefetch
+    {
+        public static async Task<ExchangePrefetchResponse> MakeRequest(HttpClient http, Item item, int minimumStock)
+        {
+            var itemName = await TranslateItemName(item);
+            Logger.Log($"ExchangePrefetch: {item.Name}, api name: {itemName}");
+            if (itemName == null)
+            {
+                Logger.Log("ExchangePrefetch: not available for this item");
+                return null;
+            }
+
+            var options = new Dictionary<string, object>
+            {
+                { "status", new Dictionary<string, string > { { "option", "online" } } },
+                { "have", new List<string> { "chaos" } },
+                { "want",  new List<string> { itemName } },
+                { "minimum", minimumStock }
+            };
+            var body = new Dictionary<string, object> { { "exchange", options } };
+            var bodySerialized = new StringContent(JsonConvert.SerializeObject(body), Encoding.UTF8, "application/json");
+
+            // TODO: league is currently hardcoded in URL
+            var response = await http.PostAsync("https://www.pathofexile.com/api/trade/exchange/Metamorph", bodySerialized);
+            Logger.Log($"ExchangePrefetch response status: {response.StatusCode}");
+            if (!response.IsSuccessStatusCode) return null;
+
+            var responseText = await response.Content.ReadAsStringAsync();
+            return JsonConvert.DeserializeObject<ExchangePrefetchResponse>(responseText);
+        }
+
+        private static IDictionary<string, string> _nameToIdMap = null;
+
+        /// <summary>Translates an item's full name into the id used by the exchange API.
+        /// For instance, "Orb of Fusing" is converted to "fuse".</summary>
+        private static async Task<string> TranslateItemName(Item item)
+        {
+            if (_nameToIdMap == null)
+            {
+                var uri = new Uri("Resources/ExchangeItemIds.json", UriKind.Relative);
```

Comment:
Remove and use the StaticItemCategories in TradeClient. It has this information, fetched from the public API. There's an example in QueryRequest.cs.